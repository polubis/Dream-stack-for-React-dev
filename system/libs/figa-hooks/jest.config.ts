/* eslint-disable */
export default {
  displayName: 'figa-hooks',
  preset: '../../jest.preset.js',
  transform: {
    '^(?!.*\\.(js|jsx|ts|tsx|css|json)$)': '@nx/react/plugins/jest',
    '^.+\\.[tj]sx?$': ['babel-jest', { presets: ['@nx/react/babel'] }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/libs/figa-hooks',
};
