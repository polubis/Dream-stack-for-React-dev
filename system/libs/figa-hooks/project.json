{
  "name": "figa-hooks",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/figa-hooks/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/figa-hooks/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "build": {
      "executor": "@nrwl/vite:build",
      "outputs": ["{options.outputPath}"],
      "defaultConfiguration": "production",
      "options": {
        "outputPath": "dist/libs/figa-hooks"
      },
      "configurations": {
        "development": {
          "mode": "development"
        },
        "production": {
          "mode": "production"
        }
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/figa-hooks/jest.config.ts",
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "codeCoverage": true
        }
      }
    }
  }
}
