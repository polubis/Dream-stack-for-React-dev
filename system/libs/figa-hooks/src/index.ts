export * from './lib/use-portal';
export * from './lib/use-toggle';
export * from './lib/use-intersection-observer';
export * from './lib/use-click-outside';
export * from './lib/use-stepper';
export * from './lib/use-scroll-y';
export * from './lib/use-client-effect';
export * from './lib/use-client-layout-effect';
export * from './lib/use-element-size';
export * from './lib/use-clipboard';
export * from './lib/use-popover';
