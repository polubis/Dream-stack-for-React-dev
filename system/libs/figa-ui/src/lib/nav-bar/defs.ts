import type { ReactNode } from 'react';

interface NavBarProps {
  className?: string;
  children: ReactNode;
  sticky?: boolean;
}

type NavBarHeaderProps = Omit<NavBarProps, 'sticky'>;

export type { NavBarProps, NavBarHeaderProps };
