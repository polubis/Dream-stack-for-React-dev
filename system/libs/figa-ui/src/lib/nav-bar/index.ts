export * from './nav-bar';
export * from './defs';
