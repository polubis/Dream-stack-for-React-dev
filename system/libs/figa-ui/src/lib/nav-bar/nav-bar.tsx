import { useScroll } from '@system/figa-hooks';
import type { NavBarHeaderProps, NavBarProps } from './defs';
import c from 'classnames';

const Header = ({ className, children }: NavBarHeaderProps) => {
  return <header className={c('nav-bar', className)}>{children}</header>;
};

const NonSticky = ({ className, children }: Omit<NavBarProps, 'sticky'>) => {
  const [state] = useScroll({ delay: 100 });

  const out = state.is === 'progress';

  return <Header className={c(className, { out })} children={children} />;
};

const NavBar = ({ sticky, ...props }: NavBarProps) => {
  const Component = sticky ? Header : NonSticky;
  return <Component {...props} />;
};

export { NavBar };
