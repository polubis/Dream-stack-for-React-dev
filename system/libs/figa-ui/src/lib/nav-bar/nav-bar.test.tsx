import { render } from '@testing-library/react';
import { NavBar } from './nav-bar';
import { ScrollReturn, useScroll } from '@system/figa-hooks';

jest.mock('@system/figa-hooks', () => ({
  useScroll: jest.fn(),
}));

describe('Navbar can be used when: ', () => {
  it('[FRAGILE] assigns classes', () => {
    (useScroll as jest.Mock).mockImplementation(
      (): ScrollReturn => [{ is: 'idle' }, { current: null }]
    );

    const { asFragment } = render(
      <NavBar className="my-class">
        <div>Content</div>
      </NavBar>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('[FRAGILE] shows navbar when user scrolled more than 20% of page', async () => {
    (useScroll as jest.Mock).mockImplementation(
      (): ScrollReturn => [
        { is: 'progress', curr: 200, prev: 180, value: 21 },
        { current: null },
      ]
    );

    const { container } = render(<NavBar>Content</NavBar>);

    expect(container.querySelector('.out')).toBeTruthy();
  });
});
