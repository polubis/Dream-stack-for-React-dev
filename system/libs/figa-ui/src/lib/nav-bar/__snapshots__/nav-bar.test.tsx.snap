// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar can be used when:  [FRAGILE] assigns classes 1`] = `
<DocumentFragment>
  <header
    class="nav-bar my-class"
  >
    <div>
      Content
    </div>
  </header>
</DocumentFragment>
`;

exports[`Navbar can be used when:  [FRAGILE] can be sticky 1`] = `
<DocumentFragment>
  <header
    class="nav-bar"
  >
    Content
  </header>
</DocumentFragment>
`;
